nixpkgs_review/github.py

```diff
@@ -83,10 +126,20 @@ def get_borg_eval_gist(self, pr: Dict[str, Any]) -> Optional[Dict[str, Set[str]]
                 raw_gist_url = (
                     f"https://gist.githubusercontent.com/GrahamcOfBorg{url.path}/raw/"
                 )
-                for line in urllib.request.urlopen(raw_gist_url):
-                    if line == b"":
-                        break
-                    system, attribute = line.decode("utf-8").split()
-                    packages_per_system[system].add(attribute)
-                return packages_per_system
+                break
+
+        if raw_gist_url is not None:
+            for line in urllib.request.urlopen(raw_gist_url):
+                if line == b"":
+                    break
+                system, attribute = line.decode("utf-8").split()
+                packages_per_system[system].add(attribute)
+            return packages_per_system
         return None
+
+    def upload_gist(self, name: str, content: str, description: str) -> Dict[str, Any]:
```

Comment:
Can we add an option to use a different token for posting gists? I have them on my bot account but would like to post as me.

Reply:
Would you want to control this by providing an environment variable like `GITHUB_GIST_TOKEN` that would have a higher precedence than the other mechanisms for setting the token?

Reply:
Thats how I do it right now. Would be great if you could add this :)

Reply:
How about adding build logs within the comment instead of a gist? https://github.com/NixOS/nixpkgs/pull/125133#issuecomment-851825296

Reply:
That seems strictly worse to me. At least in my browser (firefox), there's no dedicated HTML page for files embedded within the comment, so glancing at each log requires downloading the file and navigating to the downloads folder and opening it up.

Reply:
Fair enough.